packages/mobile/src/identity/verification.ts

```diff
@@ -222,61 +248,106 @@ export function* doVerificationFlow(withoutRevealing: boolean = false) {
-      if (Platform.OS === 'android') {
-        autoRetrievalTask = yield fork(startAutoSmsRetrieval)
-      }
-
-      yield put(setVerificationStatus(VerificationStatus.RevealingNumber))
-      ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_start)
-      yield all([
-        // Request codes for the attestations needed
-        call(
-          revealNeededAttestations,
+      if (!withoutRevealing) {
+        ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_start)
+        // Request codes for the already existing attestations if any.
+        // We check after which ones were successfull
+        const reveals: boolean[] = yield call(
+          revealAttestations,
           attestationsWrapper,
           account,
           phoneHashDetails,
-          attestations,
-          withoutRevealing
-        ),
-        // Set acccount and data encryption key in Accounts contract
-        // This is done in other places too, intentionally keeping it for more coverage
-        call(registerAccountDek, account),
-      ])
-      ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_complete)
+          attestations
+        )
+        ValoraAnalytics.track(VerificationEvents.verification_reveal_all_attestations_complete)
 
-      receiveMessageTask.cancel()
-      if (Platform.OS === 'android' && autoRetrievalTask) {
-        autoRetrievalTask.cancel()
+        // count how much more attestations we need to request
+        const attestationsToRequest =
+          status.numAttestationsRemaining - reveals.filter((r: boolean) => r).length
+
+        // check if we hit the limit for max actionable attestations at the same time
+        if (attestationsToRequest + attestations.length > MAX_ACTIONABLE_ATTESTATIONS) {
+          throw new Error(ErrorMessages.MAX_ACTIONABLE_ATTESTATIONS_EXCEEDED)
+        }
+
+        if (attestationsToRequest) {
+          yield put(setVerificationStatus(VerificationStatus.RequestingAttestations))
+          // request more attestations
+          ValoraAnalytics.track(VerificationEvents.verification_request_all_attestations_start, {
+            attestationsToRequest,
+          })
+          attestations = yield call(
+            requestAndRetrieveAttestations,
+            attestationsWrapper,
+            phoneHash,
+            account,
+            attestations,
+            attestations.length + attestationsToRequest
+          )
+          ValoraAnalytics.track(VerificationEvents.verification_request_all_attestations_complete, {
+            issuers,
+          })
+
+          // start listening for the new list of attestations
+          receiveMessageTask?.cancel()
+          issuers = attestations.map((a) => a.issuer)
+          receiveMessageTask = yield takeEvery(
+            Actions.RECEIVE_ATTESTATION_MESSAGE,
+            attestationCodeReceiver(attestationsWrapper, phoneHash, account, issuers)
+          )
```

Comment:
I notice we are listening for attestations only after all of them have been requested. Is there a risk that we might miss the first or second attestation with this approach?

What's the harm in starting to listen before we start requesting attestations?

Reply:
We do start listen to messages before, but in case the actionable attestations list is changes we resubscribe :) 

See slightly above: https://github.com/celo-org/celo-monorepo/blob/22cd8449fe457101badcde844c292b0763ee4c8e/packages/mobile/src/identity/verification.ts#L259-L262